beetsplug/metasync/itunes.py

```diff
@@ -0,0 +1,97 @@
+# This file is part of beets.
+# Copyright 2015, Tom Jaspers.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Synchronize information from iTunes's library
+"""
+from contextlib import contextmanager
+import os
+import shutil
+import tempfile
+import plistlib
+from time import mktime
+
+from beets import util
+from beets.dbcore import types
+from beets.library import DateType
+from beets.util.confit import ConfigValueError
+from beetsplug.metasync import MetaSource
+
+
+@contextmanager
+def create_temporary_copy(path):
+    temp_dir = tempfile.mkdtemp()
+    temp_path = os.path.join(temp_dir, 'temp_itunes_lib')
+    shutil.copyfile(path, temp_path)
+    try:
+        yield temp_path
+    finally:
+        shutil.rmtree(temp_dir)
+
+
+class Itunes(MetaSource):
+
+    item_types = {
+        'itunes_rating':      types.INTEGER,  # 0..100 scale
+        'itunes_playcount':   types.INTEGER,
+        'itunes_skipcount':   types.INTEGER,
+        'itunes_lastplayed':  DateType(),
+        'itunes_lastskipped': DateType(),
+    }
+
+    def __init__(self, config, log):
+        super(Itunes, self).__init__(config, log)
+
+        # Load the iTunes library, which has to be the .xml one (not the .itl)
+        library_path = util.normpath(config['itunes']['library'].get(str))
```

Comment:
Has the `as_path()` method maybe been renamed or removed? I can't seem to find it anywhere.